tests/CMakeLists.txt

```diff
@@ -24,9 +24,9 @@ else()
     CHECK_CXX_COMPILER_FLAG("-std=c++14" COMPILER_SUPPORTS_CXX14)
     CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
     if(COMPILER_SUPPORTS_CXX14)
-    	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -Wall -Wno-missing-braces")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -O3 -Wall -Wno-missing-braces")
```

Comment:
I would like to add `-fno-strict-aliasing` to these flags and then add a comment above these lines to explain why it's important to have both `-O3` and `-f-no-strict-aliasing` (i.e this way the tests can try to ensure there are not aliasing issues with the compiler when that combination of flags is applied).